src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -422,6 +424,15 @@ private static NamedTypeSymbol ValidateBuilderType(SyntheticBoundNodeFactory F,
                     !property.IsStatic &&
                     (property.ParameterCount == 0))
                 {
+                    if (property.Type != returnType)
```

Comment:
Main reason is to minimize risk. Strict equality means no un-intended fallout that we didn't have time to think through.
@cston What do you think? I'm ok either way.